File: src/main/java/ru/hlebozavod28/massak/domain/Motion.java
package ru.hlebozavod28.massak.domain;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.math.BigDecimal;
import java.sql.Timestamp;

@Data
@Entity
@NoArgsConstructor
@Table(name = "handcard_motion", schema = "workplaces")
public class Motion {
    @Id
    @GeneratedValue
    long id;
    @Column(name = "handcart_id", nullable = false)
    long handcardId;
    @Column(name = "workplace_id", nullable = false)
    long workplaceId;
    @Column(name = "sub_workplace")
    int subWorkplace;
    @CreationTimestamp
    @Column(name = "in_timestamp", nullable = false, updatable = false)
    Timestamp inTs;
    @UpdateTimestamp
    @Column(name = "out_timestamp")
    Timestamp outTs;
    @Column(name = "product_code")
    int productCode;
    @Column(name = "amount")
    BigDecimal amount;

    public Motion(long workplaceId, long handcardId, int productCode) {
        this.handcardId = handcardId;
        this.workplaceId = workplaceId;
        this.productCode = productCode;
    }
}

File: src/main/java/ru/hlebozavod28/massak/domain/Motion.kt
package ru.hlebozavod28.massak.domain

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.math.BigDecimal
import java.sql.Timestamp
import javax.persistence.*

@Entity
@Table(name = "handcart_motion", schema = "workplaces")
class Motion(
    @Column(name = "workplace_id", nullable = false)
    var workplaceId: Long ,
    @Column(name = "handcart_id", nullable = false)
    var handcartId: Long ,
    @Column(name = "product_code")
    var productCode: Int ,
    @Column(name = "sheets")
    var sheets: Int
) {
    @Id
    @GeneratedValue
    var id: Long = 0

    @Column(name = "sub_workplace")
    var subWorkplace = 0

    @CreationTimestamp
    @Column(name = "in_timestamp", nullable = false, updatable = false)
    var inTs: Timestamp? = null

    @UpdateTimestamp
    @Column(name = "out_timestamp")
    var outTs: Timestamp? = null

    @Column(name = "amount")
    var amount: BigDecimal? = null

    @Column(name = "defect_count")
    var defectCount: BigDecimal? = null

    @Column(name = "deleted")
    var deleted = false
}